src/material/datepicker/calendar.ts

```diff
@@ -393,7 +393,27 @@ export class MatCalendar<D> implements AfterContentInit, AfterViewChecked, OnDes
   }
 
   ngOnChanges(changes: SimpleChanges) {
-    const change = changes['minDate'] || changes['maxDate'] || changes['dateFilter'];
+    let change = undefined;
+
+    for (let key in changes) {
```

Comment:
The use of the for loop that breaks when it finds a change is a little hard to follow. Would prefer to see more straightforward logic here, e.g. something more like

```
const minDateChanged = !this._dateAdapter.sameDate(changes['minDate'].previousValue, changes['minDate'].currentValue) && !changes['minDate'].firstChange;
const minDateChanged = !this._dateAdapter.sameDate(changes['maxDate'].previousValue, changes['maxDate'].currentValue) && !changes['maxDate'].firstChange;
const dateFilterChanged = changes['dateFilter'] && !changes['dateFilter'].firstChange

if (minDateChanged || maxDateChanged || dateFilterChanged) {
  ...
}

```

Reply:
Okay, let's not use the for-loop approach.